data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/PeerListProviderFactory.java

```diff
@@ -27,30 +18,7 @@ public PeerListProvider createProvider(final PluginSetting pluginSetting) {
 
         final PluginMetrics pluginMetrics = PluginMetrics.fromPluginSetting(pluginSetting);
 
-        switch (discoveryMode) {
-            case DNS:
-                final String domainName = pluginSetting.getStringOrDefault(PeerForwarderConfig.DOMAIN_NAME, null);
-                Objects.requireNonNull(domainName, String.format("Missing '%s' configuration value",PeerForwarderConfig. DOMAIN_NAME));
-                Preconditions.checkState(validateEndpoint(domainName), "Invalid domain name: %s", domainName);
-
-                final DnsAddressEndpointGroup endpointGroup = DnsAddressEndpointGroup.builder(domainName)
-                        .ttl(MIN_TTL, MAX_TTL)
-                        .build();
-
-                return new DnsPeerListProvider(endpointGroup, pluginMetrics);
-            case STATIC:
-                final List<String> endpoints = (List<String>) pluginSetting.getAttributeOrDefault(PeerForwarderConfig.STATIC_ENDPOINTS, null);
-                Objects.requireNonNull(endpoints, String.format("Missing '%s' configuration value", PeerForwarderConfig.STATIC_ENDPOINTS));
-                final List<String> invalidEndpoints = endpoints.stream().filter(endpoint -> !this.validateEndpoint(endpoint)).collect(Collectors.toList());
-                Preconditions.checkState(invalidEndpoints.isEmpty(), "Including invalid endpoints: %s", invalidEndpoints);
-
-                return new StaticPeerListProvider(endpoints, pluginMetrics);
-            default:
-                throw new UnsupportedOperationException(String.format("Unsupported discovery mode: %s", discoveryMode));
```

Comment:
Nit: If we want to be specific about exception consider adding a method in DiscoveryMode class to get the discoveryMode and throw UnsupportedDiscoveryModeException if invalid discoveryMode is passed. By default IllegalArgumentException will be thrown if the specified enum type has no constant with the specified name or NullPointerException if enum type is null.

Are we handling UnsupportedOperationException ?

Reply:
Looks like the previous behavior was throwing IllegalArgumentException, so we can keep the same. Just a thought here.

Reply:
This original behavior should no longer happen. The original code required that you add the new enum and add code to this conditional. Now, in order to add a new mode you must add the create function. So this should no longer happen.

I did keep the original behavior which throws either:

* `NullPointerException` - No Discovery Mode provided
* `IllegalArgumentException` - The string value provided does not map to the enum

I think it could be valuable to combine these into a single exception like `InvalidDiscoveryModeException`. The reason could indicate the exact problem - missing or invalid.

Reply:
I created #737 so that we can circle back on this in a later commit.